modin/engines/base/frame/data.py

```diff
@@ -2226,10 +2226,7 @@ def to_pandas(self):
         """
         df = self._partition_mgr_cls.to_pandas(self._partitions)
         if df.empty:
-            if len(self.columns) != 0:
-                df = pandas.DataFrame(columns=self.columns)
-            else:
-                df = pandas.DataFrame(columns=self.columns, index=self.index)
+            df = pandas.DataFrame(columns=self.columns, index=self.index)
```

Comment:
Could you please explain what this change does? It looks okay, but I'm wondering why the original code was like it was, and why this was needed to fix the tests...

Reply:
@vnlitvinov len (columns) always! = 0, because there the column '__reduced__' is written in the case of an empty dataframe, because of which the length will not be equal to zero. And we will always have a default index.